deploy/docker/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM nucypher/rust-python:3.8.16
+FROM nucypher/rust-python:3.12.0
```

Comment:
here is the newly published matching base image with the rust build toolchain:

https://hub.docker.com/layers/nucypher/rust-python/3.12.0/images/sha256-dc362be5ca96b53221238812a1648cc2a93a2f12777515efa8da33052748edae?context=repo